src/sage/combinat/combinat.py

```diff
@@ -2963,28 +2993,88 @@ def bell_polynomial(n: Integer, k: Integer):
         [[], [1], [3], [1]]
         [[], [1], [3, 4], [6], [1]]
 
+    Further checks for :issue:`37727`::
+
+        sage: # needs sage.combinat sage.arith 
+        sage: bell_polynomial(0, 0)
+        1
+        sage: bell_polynomial(0, 0, True)
+        1
+        sage: bell_polynomial(1, 1)
+        x0
+        sage: bell_polynomial(2, 2, True)
+        x0^2
+        sage: bell_polynomial(5)
+        x0^5 + 10*x0^3*x1 + 15*x0*x1^2 + 10*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
+        sage: sum(bell_polynomial(5, k) for k in range(6))
+        x0^5 + 10*x0^3*x1 + 15*x0*x1^2 + 10*x0^2*x2 + 10*x1*x2 + 5*x0*x3 + x4
+        sage: bell_polynomial(5, None, True)
+        x0^5 + 4*x0^3*x1 + 3*x0*x1^2 + 3*x0^2*x2 + 2*x1*x2 + 2*x0*x3 + x4
+        sage: sum(bell_polynomial(5, k, True) for k in range(6))
+        x0^5 + 4*x0^3*x1 + 3*x0*x1^2 + 3*x0^2*x2 + 2*x1*x2 + 2*x0*x3 + x4
+        sage: bell_polynomial(0).parent()
+        Univariate Polynomial Ring in x0 over Integer Ring
+
+    We verify that Bell polynomials equal their power series definition:
+    - if `f(z) = \sum_{n\geq1} x_{n-1} z^n/n!` then
+      `\exp(f) = \sum_{n\geq0} B_n z^n/n!` and
+      `f^k/k! = \sum_{n\geq k} B_{n, k} z^n/n!`
+    - if `g(z) = \sum_{n\geq1} x_{n-1} z^n` then
+      `1/(1-f) = \sum_{n\geq0} \hat B_n z^n` and
+      `f^k = \sum_{n\geq k} \hat B_{n, k} z^n`
+    (see reference).
```

Comment:
This entire block should be in the description with simply a reference back to that block. It doesn't have to be explicit, but something like
```
    We verify the description of the (exponential) Bell polynomials
    using the functions `f` and `g` given above::
```